### PR TITLE
C++arrays

### LINKED ISSUE #4364: [Edit] C++ Arrays
### Type of Edit (select all that apply)

Clarify unclear content and/or terms, Add new content (definitions, codeblocks, etc.)

### Description (optional)

In the subsection "Array indices" of this entry it is described how to initialize an array, how to change initialized elements and how to access initialized elements. It is also pointed out that the size of the array cannot be changed during runtime.

What needs to be added is in my opinion
- the recommendation to not access uninitialized elements. E.g. if we have int arr[5] = {0, 1, 2}, then the elements at index 3 and 4 are not initialized and one should not access them before they're initialized, because there can be anything stored inside.
- making the programmer aware of that even if an array arr has length n, you can access and modify with arr[n+1] which leads to undefined behaviour and can e.g. overwrite other variables. This one is especially important since I think that beginners could have some trouble to manage the array length information in their programs which could lead them to use index out of bounds. Since there is not an index out of bounds exception, not knowing this fact could lead them to have hard times debugging their program.

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [ ] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by JamesMasterton**
### Description

Added section to C++ Arrays markdown file (content/cpp/concepts/arrays/arrays.md) about uninitialized elements.
Done as part of the "Contributing to the Codecademy Docs Repository" project on the "Learn Git & GitHub" course.

### Issue Solved
Closes #4364

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/cpp/concepts/arrays/arrays.md [hunk 1]
```diff
@@ -59,3 +59,13 @@ std::cout << vowels[0];
 In the case above, an array of `chars` was initialized with all the vowels, and then the first element in the array at index `0` was printed out. Then the element at index `0` was modified by being assigned a new value of `'r'`, which was then printed out.
```
Comment (mohitO12-dev) [inline content/cpp/concepts/arrays/arrays.md:59]: Just to make the paragraph more readable add this 

In the case above, an array of **`chars`** was initialized with all the vowels, and then the first element in the array at index **`0`** was printed out. Subsequently, the element at index **`0`** was modified by assigning it a new value of **`r`**, which was then printed out.

These changes help to refine the flow and clarity of the sentence.

File: content/cpp/concepts/arrays/arrays.md [hunk 2]
```diff
@@ -59,3 +59,13 @@ std::cout << vowels[0];
...
 
 Arrays in C++ have a fixed size, meaning elements can not be added or removed once the array has been created. Only existing elements may be modified without changing the total size or shape of the structure.
```
Comment (mohitO12-dev) [inline content/cpp/concepts/arrays/arrays.md:61]:
```suggestion
Arrays in C++ have a fixed size, meaning elements cannot be added or removed once the array has been created. Only existing elements may be modified without changing the total size or shape of the structure.
```

File: content/cpp/concepts/arrays/arrays.md [hunk 3]
```diff
@@ -59,3 +59,13 @@ std::cout << vowels[0];
...
+
+## Uninitialized Elements
+
+Array elements that are not assigned a value when the array is created are known as uninitialized, and should not be accessed during runtime. For example:
```
Comment (mohitO12-dev) [inline content/cpp/concepts/arrays/arrays.md:65]: Array elements that are not assigned a value when the array is created are known as uninitialized and should not be accessed during runtime. For example:"

I removed the comma after "uninitialized" to streamline the sentence structure.

File: content/cpp/concepts/arrays/arrays.md [hunk 4]
```diff
@@ -59,3 +59,13 @@ std::cout << vowels[0];
...
+
+```cpp
+int arr[5] = {0, 1, 2}
```
Comment (mohitO12-dev) [inline content/cpp/concepts/arrays/arrays.md:68]: Added ';' for proper syntax in c++ array.
```suggestion
int arr[5] = {0, 1, 2};
```

**Review by mohitO12-dev — APPROVED** [review]

**Review by mohitO12-dev — APPROVED** [review]

**Review by mohitO12-dev — APPROVED** [review]
Approved.

**Review by Sriparno08 — APPROVED** [review]
Looks good 👍

@JamesMasterton From the next time, please make sure you start working on an issue after you get assigned to it. It prevents confusion and ensures that your hard work is rewarded.

**Pull request merged**